io/disk/dbench.py

```diff
@@ -89,24 +98,36 @@ def setUp(self):
         process.run('patch -p1 < %s' % self.get_data(patch), shell=True)
         process.run('./configure')
         build.make(self.sourcedir)
-        self.dirs = self.disk
         if self.disk is not None:
             if self.disk in disk.get_disks():
                 if raid_needed:
-                    raid_name = '/dev/md/mdsraid'
+                    raid_name = '/dev/%s' % self.md_name
                     self.create_raid(self.disk, raid_name)
                     self.raid_create = True
                     self.disk = raid_name
-                    self.dirs = self.disk
                 if lv_needed:
                     self.disk = self.create_lv(self.disk)
                     self.lv_create = True
-                    self.dirs = self.disk
                 if fstype:
-                    self.dirs = self.workdir
-                    self.create_fs(self.disk, self.dirs, fstype)
+                    self.create_fs(self.disk, fstype)
                     self.fs_create = True
 
+    def pre_cleanup(self):
+        umount_dir = "umount -f %s" % self.mountpoint
+        process.system(umount_dir, shell=True, ignore_status=True)
+        delete_lv = "lvremove -f /dev/mapper/avocado_vg-avocado_lv"
+        process.system(delete_lv, shell=True, ignore_status=True)
+        delete_vg = "vgremove -f avocado_vg"
+        process.system(delete_vg, shell=True, ignore_status=True)
+        delete_rd = 'mdadm --stop /dev/%s' % self.md_name
+        process.system(delete_rd, shell=True, ignore_status=True)
+
```

Comment:
@abdhaleegit why are we using hard coded values, and why not the libraries?
I understand tests are failing even with existing libraries some time but lets standardize the libraries itself instead adding this part.
there are any other test cases of disks are failing due to unclean mount points and fs on them.
So I suggest to work on libs, so that same can be leverage for other tests too.

Reply:
Also before the test start we are not knowing the disk obj, or lv object or raid obj.. so is the reason I did not use the libraries... this is only pure pre cleanup.. unlike teardown where we have initiated objs

Reply:
this takes care of all the cases in case if test was interrupted in mid..  vgs, lvs and rd  left unclean.. so to handle this case this hardcode is added